src/Domain/Regulation/Enum/HeavyweightMaxWeightEnum.php

```diff
@@ -0,0 +1,15 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Domain\Regulation\Enum;
+
+enum HeavyweightMaxWeightEnum: string
+{
+    case T3_5 = '3.5t';
+    case T7_5 = '7.5t';
+    case T19 = '19t';
+    case T26 = '26t';
+    case T32 = '32t';
+    case T44 = '44t';
+}
```

Comment:
Que penses-tu de créer un enum de type float ?

Cela permet de continuer à stocker un float en DB, ce qui semble logique car le poids est bien un nombre de tonnes

L'enum aura alors juste un rôle de validation / restriction des valeurs

```suggestion
enum HeavyweightMaxWeightEnum: float
{
    case T3_5 = 3.5;
    case T7_5 = 7.5;
    case T19 = 19;
    case T26 = 26;
    case T32 = 32;
    case T44 = 44;
}
```

Reply:
j'ai essayé mais l'enum n'accepte que le type int ou string ... mais je vais être obligé de trouver une solution car en l'état je peux pas sauvegarder comme ça

Reply:
Ah OK je comprends

Peut-être alors garder un "simple" `float`, mais utiliser une `<constraint name="Choice">` dans `SaveVehicleSetCommand.xml` ?

Ça nécessite juste de "dupliquer" à deux endroits la liste des choix, une fois là et une fois dans le formType, mais ce n'est pas si grave. On le fait déjà pour SaveMeasureCommand::type par exemple (noEntry, speedLimitation)